docs/examples/wasm/webpack/src/secure-cell.js

```diff
@@ -0,0 +1,407 @@
+
+    const outputBoxTokenProtect = document.getElementById('secure-cell-token-protect-output')
+    const errorBoxTokenProtect = outputBoxTokenProtect.querySelector('div.error-box')
+
+    const ciphertextContextImprint = document.getElementById('secure-cell-context-imprint-ciphertext')
+    const ciphertextContextImprintArea = ciphertextContextImprint.querySelector('textarea')
+    const ciphertextContextImprintLength = ciphertextContextImprint.querySelector('span.size-hint')
+
+    const outputBoxContextImprint = document.getElementById('secure-cell-context-imprint-output')
+    const errorBoxContextImprint = outputBoxContextImprint.querySelector('div.error-box')
+
+    function adjustTextAreaRows(area) {
+        // Non-BMP characters? Grapheme clusters? Emoji? Who uses *that*?
+        const rows = area.value.split('\n').reduce(((rows, row) => {
+            return rows + Math.floor((row.length + area.cols - 1) / area.cols)
+        }), 0)
+        if (rows <= 5 && area.rows < rows) {
+            area.rows = rows
+        }
+    }
+
+    function massageErrorMessage(error) {
+        let message = error.toString()
+        const colonIndex = message.lastIndexOf(': ')
+        if (colonIndex != -1) {
+            message = message.slice(0, colonIndex) + ':<br>' + message.slice(colonIndex + 2)
+        }
+        return message
+    }
+
+    function encryptSealWithKey() {
+        try {
+            const decodedKey = Base64ToBytes(symmetricKeyInput.value)
+            const encodedContext = StringToUTF8Array(contextArea.value)
+            const encodedPlaintext = StringToUTF8Array(plaintextArea.value)
+
+            const cell = themis.SecureCellSeal.withKey(decodedKey)
+            const ciphertext = cell.encrypt(encodedPlaintext, encodedContext)
+
+            ciphertextSealArea.value = BytesToBase64(ciphertext)
+            ciphertextSealLength.textContent = `${ciphertext.length} bytes`
+            adjustTextAreaRows(ciphertextSealArea)
+
+            errorBoxSeal.textContent = ''
+            errorBoxSeal.classList.add('hidden')
+        }
+        catch (error) {
+            ciphertextSealArea.value = ''
+            ciphertextSealLength.textContent = '0 bytes'
+
+            errorBoxSeal.innerHTML = massageErrorMessage(error)
+            errorBoxSeal.classList.remove('hidden')
+        }
+    }
+
+    function decryptSealWithKey() {
+        try {
+            const decodedKey = Base64ToBytes(symmetricKeyInput.value)
+            const encodedContext = StringToUTF8Array(contextArea.value)
+            const decodedCiphertext = Base64ToBytes(ciphertextSealArea.value)
+
+            const cell = themis.SecureCellSeal.withKey(decodedKey)
+            const plaintext = cell.decrypt(decodedCiphertext, encodedContext)
+
+            plaintextArea.value = UTF8ArrayToString(plaintext)
+            plaintextLength.textContent = `${plaintext.length} bytes`
```

Comment:
wow how JS can do ) situation when hasn't used js several years and meet new features